tests/apidoc_request_encoding_test.go

```diff
@@ -0,0 +1,29 @@
+
+package tests
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+)
+
+func TestRequestEncodingFails(t *testing.T) {
+	deployment := Deploy(t, "request_encoding", b.BlueprintAlice)
+	defer deployment.Destroy(t)
+	unauthedClient := deployment.Client(t, "hs1", "")
+	// sytest: POST rejects invalid utf-8 in JSON
+	t.Run("POST /login can login as user", func(t *testing.T) {
+		res := unauthedClient.MustDo(t, "POST", []string{"_matrix", "client", "r0", "login"}, json.RawMessage(`{
+			"test": "a' . chr(0x81) . '"
```

Comment:
I don't think `"a' . chr(0x81) . '"` is valid Go, is it? I know that's what we're using in on the Perl side for SyTest, but I think Go uses different syntax and functions for this.

Reply:
But we just need an invalid json over here right? Why do we need to think about syntax 🤔

Reply:
You make a good argument 🙂 ... but hilariously moving this to Go _does_ create valid JSON, since it literally encodes to the string "`a' . chr(0x81) . '`" 🤣 

I think the intent is to somehow jam a [Unicode Control Character](https://www.fileformat.info/info/unicode/char/0081/index.htm) in there, so we'd need to do something like:

```
"test": "a` + string(0x81) + `"
```

Reply:
Makes sense, sorry for the silly arguements😅 Amending it right away